doc/samples/variants/demo.rst

```diff
@@ -0,0 +1,95 @@
+.. _sidewalk_demo:
+
+Sensor monitoring
+#################
+
+This sample demonstrates how a temperature sensor communicates over the secure Sidewalk network, which is a simple Sidewalk-based IOT use case.
+It consists of an edge device (a sensor).
+An application server (cloud backend with web UI) compatible with this sample is provided in `Amazon Sidewalk Sample IoT App`_ repository.
+
+
+User Interface
+**************
+
+.. include:: include_user_interface_common.txt
+
+Button 1-4 (short press):
+   Send button action message. This action works after application successfully sends capability message.
+
+LED 1-4:
+   Notify LED action message. This action works after application successfully sends capability message.
+
+Configuration
+*************
+
+.. note::
+    To build Sidewalk end device as sensor monitoring application
+    use ``OVERLAY_CONFIG="overlay-demo.conf"``.
+
+The Sidewalk sensor monitoring demo application supports the following configurations:
+
+* ``CONFIG_SID_END_DEVICE_NOTIFY_DATA_PERIOD_MS`` -- Notify period of sensor monitoring end device app in milliseconds.
+
+For more configuration option see :ref:`sidewalk_end_device_configuration`.
+
+Source file setup
+*****************
+
+The application consists of two source files:
+
+* :file:`app.c` (:file:`app.h`) -- The main application file starts the TX and RX thread, assigns button actions to the board's buttons, and adds received messages to the received message queue.
+* :file:`app_sensor.c`, :file:`app_button.c`, :file:`app_led.c` -- The files implement interface between the hardware and sample application.
+* :file:`app_rx.c` (:file:`app.h`) -- RX thread reads messages form the received messages queue, deserializes them, and triggers actions.
+* :file:`app_tx.c` (:file:`app.h`) -- TX thread sends messages from the device to cloud.
+  Depending on the actual state of the device, it sends capability messages or sensor data notifications.
+
+  .. uml::
+      :caption: Application TX thread state machine
+
+      [*] -> Init
+      Init --> Notify_Capacity : time_syc_success
+      Notify_Capacity --> Notify_Data : capacity_success
+      Notify_Data --> Notify_Capacity : time_syc_fail
+      Notify_Data --> Notify_Data : action_response
+      Notify_Data --> Notify_Data : action_notify
+
+Building and running
+********************
+
+.. include:: include_building_and_running.txt
+
+Testing
+*******
+
+.. note::
+   For the full installation guidelines and the application overview, refer to the `Amazon Sidewalk Sample IoT App`_ repository.
```

Comment:
tbc - Why is the user referred to installation guide and the app overview in the testing section? Shouldn't this reference appear earlier?

Reply:
Is there any point in repeating it here then? Unless we're referring user to some other testing instructions

Reply:
Noted - thank you!